### PR TITLE
Output android translations as Kotlin Multiplatform project

### PR DESCRIPTION
**Pull request by sleeyax**
Migrated from the regular android project output to a multiplatform project. Basically, `android/main` now becomes `android/androidMain` and `android/commonMain` has been added in order to deliver common translations (in kotlin class format) to other multiplatform projects.

This PR is not intended as a breaking change. Nothing should change by this PR besides added support for Kotlin Multiplatform or KMM projects that wish to add translations to their `commonMain` sourceset.

Marked as draft because I need to run more tests and adjust the publishing configuration.

### CODE REVIEW

File: scripts/build-kmm.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const dedup = deduplicate(defaultTranslation);
+  Object.keys(dedup).forEach(key => stringClass += `  val ${key}: String,\n`);
+  stringClass += ')';
+  fs.writeFileSync(path.join(process.cwd(), rootPath, 'Strings.kt'), stringClass);
+}
+
+function writeLocales() {
+  let stringClass = `package ${packageName}\n\n`;
+  stringClass += 'object Locales {\n';
+  const locales = new Set(files.map(file => file.split('-')[0]));
```
Comment (TheBeastLT) [inline scripts/build-kmm.js:34]: Why are you taking only the country part of it? You should use the full language code, as you'll loose some languages (ie pt-PT and pt-BR)?

Comment (sleeyax) [reply scripts/build-kmm.js]: I was like "I need to split this to get the locale only", but that you're right that doesn't make sense. It should be the full language code indeed.

**Review by sleeyax — COMMENTED** [review]
Curious, is there a particular reason to use node 12? (which is EOL since 30 April 2022 btw)

**Comment by TheBeastLT** [comment]
> Curious, is there a particular reason to use node 12? (which is EOL since 30 April 2022 btw)

For build scripts probably there's no particular reason, but this version is used when running streaming server on android.

**Comment by sleeyax** [comment]
> > Curious, is there a particular reason to use node 12? (which is EOL since 30 April 2022 btw)
> 
> For build scripts probably there's no particular reason, but this version is used when running streaming server on android.

But doesn't this repo only contain build scripts and no code that's ever to be executed by the streaming server? If so, there's no particular reason to stick to Node 12. I'll stick to it for now for consistency reasons, but perhaps we should consider upgrading to a newer version (in a new PR) so we can benefit from new language features that would make the code more readable and easier to maintain.

**Comment by sleeyax** [comment]
I've marked this PR as ready. Did some profiling and the implementation with [lyricist](https://github.com/adrielcafe/lyricist) is stable in a KMM (android) app, just about the same as a custom implementation with `strings.xml`. Suggestions or additional feedback are welcome.

**Review by TheBeastLT — APPROVED** [review]

**Comment by jaruba** [comment]
@sleeyax 

> I'll stick to it for now for consistency reasons

this is pretty much the reason, everyone sticks to their tasks and we only upgrade things when necessary as we have way too many repos to maintain

**Pull request merged**